### PR TITLE
Always use userspace IOAPIC and MSI for virtio devices 

### PR DESCRIPTION
**Pull request by rbradford**
Remove lots of code that wasn't being widely used.

Fixes: #511
Fixes: #282

### CODE REVIEW

File: vmm/src/vm.rs [hunk 1]
```diff
@@ -145,6 +145,9 @@ pub enum Error {
 
     /// Error from CPU handling
     CpuManager(cpu::Error),
+
+    /// apability missing
```
Comment (sboeuf) [inline vmm/src/vm.rs]: s/apability/capability/

**Comment by sboeuf** [comment]
@rbradford as a follow up PR (if we merge this one), we should add support for removing the IOAPIC when nothing relies on INTx.
Maybe we could compile it out too?

**Comment by rbradford** [comment]
> @rbradford as a follow up PR (if we merge this one), we should add support for removing the IOAPIC when nothing relies on INTx.
> Maybe we could compile it out too?

Yeh I intentionally left that possibility open by returning an `Option` in the accessor.

File: vmm/src/vm.rs (hunk 1 above)
Comment (rbradford) [reply vmm/src/vm.rs]: Oops, thanks.

**Comment by rbradford** [comment]
The MSI and TSC_DEADLINE_TIMER caps were added to KVM ~8 years ago. SPLIT_IRQCHIP was added 4 years ago.

**Comment by sboeuf** [comment]
@rbradford 
>The MSI and TSC_DEADLINE_TIMER caps were added to KVM ~8 years ago. SPLIT_IRQCHIP was added 4 years ago.

Yes that's why I think this PR makes sense :)

**Comment by sameo** [comment]
Nice, we should have done that a long time ago. Thanks @sboeuf @rbradford.

**Pull request merged**